### DISCUSSION TITLE
Select several columns and plot them

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2424] [Q&A] [answered]
Here's a need I often have. I have a sheet with lots of columns. The key column is already marked. I want to select three of them manually and plot them. Currently all ways I have of doing this suck, and basically revolve around hiding all the other columns. I wish I had a fast way to just mark a few columns to have them plotted. Is there currently a way to do that?

**Comment by saulpw** [comment]
There's not; we don't currently have a way to select columns on the main sheet.   What if we overloaded `g.` on the Columns Sheet to graph selected columns on the source sheet?

**Reply by cool-RR** [reply]
Interesting idea. I did a make-believe session right now and it seems convient. Shall I open a feature request for it?

**Reply by saulpw** [reply]
Try this in your .visidatarc (I just copied the execstr for `plot-numerics` and changed `sheet` to `source` and `nonKeyVisibleCols` to `selectedRows`):

```
ColumnsSheet.addCommand('g.', 'plot-source-selected', 'vd.push(GraphSheet(source.name, "graph", source=source, sourceRows=source.rows, xcols=source.keyCols, ycols=numericCols(selectedRows)))', 'plot a graph of all selected columns vs key columns on source sheet')
```

Maybe with `vd.replace` instead, if you don't want to come back to the ColumnsSheet.

**Reply by cool-RR** [reply]
Tried it, I get ` AttributeError: 'list' object has no attribute 'name'`.

**Reply by saulpw** [reply]
Okay, so, change `source` to `source[0]` throughout.  Sorry I didn't test it (haven't tested this either, but I'd bet we're within striking distance :)

**Reply by cool-RR** [reply, marked as answer]
This works great! Thank you. 

Here's the corrected script for anyone who's interested:

```python
ColumnsSheet.addCommand('g.', 'plot-source-selected', 'vd.push(GraphSheet(source[0].name, "graph", source=source[0], sourceRows=source[0].rows, xcols=source[0].keyCols, ycols=numericCols(selectedRows)))', 'plot a graph of all selected columns vs key columns on source sheet')
```

**Reply by cool-RR** [reply]
Follow up 3 weeks later: I've been using this script hundreds of times per day and it's a great time saver.

**Reply by anjakefala** [reply]
We love to hear that! It's actually now in VisiData proper. =) Thank you for the feature request!

**Reply by anjakefala** [reply]
Here is the commit: https://github.com/saulpw/visidata/commit/5723b4bd94940f18c2c506c5e97938b25e377cd2. It will be in the next release.
